exporters/ostream/include/opentelemetry/exporters/ostream/log_exporter.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "opentelemetry/nostd/type_traits.h"
+#include "opentelemetry/sdk/logs/exporter.h"
+#include "opentelemetry/sdk/logs/log_record.h"
+#include "opentelemetry/version.h"
+
+#include <iostream>
+#include <sstream>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * The OStreamLogExporter exports logs through an ostream (default set to std::cout)
+ */
+class OStreamLogExporter final : public opentelemetry::sdk::logs::LogExporter
+{
+public:
+  /**
+   * Create an OStreamLogExporter. This constructor takes in a reference to an ostream that the
+   * Export() method will send log data into. The default ostream is set to stdout.
+   */
+  explicit OStreamLogExporter(std::ostream &sout = std::cout) noexcept;
+
+  /*********************************** Helper functions ************************/
+
+  /*
+    print_value is used to print out the value of an attribute within a vector.
+    These values are held in a variant which makes the process of printing them much more
+    complicated.
+  */
+
+  template <typename T>
+  void print_value(const T &item)
```

Comment:
if "print_value" and "OwndedValueVisitor" are helper functions, why not just put them into the `.cc` file?

Yes, you have to additionally pass sout_, but you'd still get your template-value-static-do-the-right-thing-with-my-type behavior you want, and we won't have a bunch of weak symbols to resolve later when linking *or* public methods that shouldn't be public.


Something like:

```
template <typename T>
void print_value(const T& item, std::ostream& sout) { ... }
template <typename T>
void print_value(const std::vector<T>& item, std::ostream& sout) { ...}
template <typename T>
void print_value(sdk::common::OwnedAttributeVisitor& item, std::ostream& sout) { ...}
```

Reply:
Mmk yeah, sorry, they were supposed to be private - they're now moved to the `.cc` file